proposals/0000-down_ranked_exists.rst

```diff
@@ -0,0 +1,462 @@
+
+.. _`#473`: https://github.com/ghc-proposals/ghc-proposals/pull/473
+.. _`#81`:  https://github.com/ghc-proposals/ghc-proposals/blob/master/proposals/0081-forall-arrow.rst
+.. _`#281`: https://github.com/ghc-proposals/ghc-proposals/blob/master/proposals/0281-visible-forall.rst
+.. _`#281rd`: https://ghc-proposals.readthedocs.io/en/latest/proposals/0281-visible-forall.html
+.. _`#378`: https://github.com/ghc-proposals/ghc-proposals/blob/master/proposals/0378-dependent-type-design.rst
+.. _`#378rd`: https://ghc-proposals.readthedocs.io/en/latest/proposals/0378-dependent-type-design.html
+
+
+Motivation
+----------
+
+Richly typed programming invariably uses its share of existential types, and this proposal makes it vastly easier to work with existentials. 
+
+Currently, every existential must be encoded using its own datatype, which is laborious. Furthermore, packing and unpacking these datatypes must be done by hand, which is cluttersome.
+
+UpRanked Existentials
+~~~~~~~~~~~~~~~~~~~~~
+
+Alternative Proposal is "First-class existential types" `#473`_ .
+
+#473 suggests to add *de facto* **UpRanked Existential**
+
+- **UpRanked Existential** rule: Any N-Ranked ``forall`` *type_variable* is INSTEAD (N+1)-Ranked ``exists`` *type_variable* 
+::
+
+  --UpRanked Existentials:
+  cmpg :: forall a. a -> a ->(forall b. b -> b -> Bool) -> Bool
+  cmpg x y f = f x y
+
+  -- same as
+  cmpg :: forall a. a -> a ->((exists b. b -> b) -> Bool) -> Bool
+
+  -- same as
+  cmpg :: (exists a. a -> a ->((exists b. b -> b) -> Bool)) -> Bool
+
+Unfortunately, working with High-Ranked Types is complicated in Haskell and it's Core.
+
+DownRanked Existentials
+~~~~~~~~~~~~~~~~~~~~~~~
+
+This Proposal suggest to add the opposite (of #473) : **DownRanked Existential**
+
+- **DownRanked Existential** rule: Any N-Ranked ``forall`` (or ``exists``) *type_variable* is ALSO (N-1)-Ranked ``exists`` *type_variable* 
+::
+
+  --DownRanked Existentials:
+  cmpg2 :: forall a. a -> a ->(forall b. b -> b -> (b, Bool)) -> (a, Bool)
+  cmpg2 x y f = f x y
+
+  -- same as
+  cmpg2 :: forall a. exists b. a -> a ->(forall b. b -> b -> (b, Bool)) -> (b, Bool)
+
+
+The Main rule: *if* ∀a: f a *then* ∃b: ∀a, a ≡ b : f b
+
+Second rule: *if* ∃a: f a *then* ∃b: ∃a, a ≡ b : f b
+
+This is the core idea of this Dependent existential type from Higher-Ranked (Exactly N+1 Ranked) ``forall`` / ``exists`` type variables with `a ≡ b` equality condition.
+
+*Note: using same keyword "exists" for both UpRanked and DownRanked Existential Quantifiers is incompatible and inconsistent idea*
+
+
+Proposed Change Specification
+-----------------------------
+
+Main and second rules give us unique proprieties of DownRanked existentials.
+
+Roles
+~~~~~
+
+1. Boxing type variable into a Data declaration 
+::
+
+  -- hidden conventional existential GHC type
+  data Box = forall a. MkBox a
+
+  -- open existential type
+  data exists a. Ex = forall a. MkEx { unEx :: a }  -- NEW!
```

Comment:
Why is `Ex` quantified with `exists a.` but has no free `a`?
Why do you shadow `a` on the right hand side?

Reply:
1) We don't shadow right `a`, but we capture right `a`
2) We have 2 alternative options: `exists a. Box` and `exists a. Box a`. I'm not against both of them. And I'm in favor in second one. 
Main criteria to chose - to preserve "absorption" rule.